### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by per1234**
This adds end to end testing of the tool to verify that the finished outputs are as expected.

### CODE REVIEW

File: test/test_all.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    working_dir_path = pathlib.Path(working_dir)
+    configuration = {
+        "BaseDownloadUrl": "http://www.example.com/libraries/",
+        "LibrariesFolder": working_dir_path.joinpath("libraries").as_posix(),
+        "LogsFolder": working_dir_path.joinpath("logs").as_posix(),
+        "LibrariesDB": working_dir_path.joinpath("libraries_db.json").as_posix(),
+        "LibrariesIndex": working_dir_path.joinpath("libraries", "library_index.json").as_posix(),
+        "GitClonesFolder": working_dir_path.joinpath("gitclones").as_posix(),
+        # I was unable to get clamdscan working in the GitHub Actions runner, but the tests should pass with this set to
+        # False when run on a machine with ClamAV installed.
```
Comment (rsora) [inline test/test_all.py:49]: As we were discussing in our chats, being this application something that has to run on a well defined environment, could make sense to include in this repo a `Dockerfile` with a replica of the destination linux environment, included all the tools that this engine uses (like ClamAV)

File: test/test_all.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        quoted_cmd = []
+        for token in cmd:
+            quoted_cmd.append(f'"{token}"')
+        cli_full_line = '"{}" {}'.format(executable_path, " ".join(quoted_cmd))
+        run_context = invoke.context.Context()
+        # It might happen that we need to change directories between drives on Windows,
+        # in that case the "/d" flag must be used otherwise directory wouldn't change
+        cd_command = "cd"
+        if platform.system() == "Windows":
+            cd_command += " /d"
```
Comment (rsora) [inline test/test_all.py:204]: Same consideration as above, with a container that replicates the production environment we could get rid of this "OS dependant" code

**Review by rsora — APPROVED** [review]
The code is very ok, we could merge it and then address my comments about having a test container in a separate PR :+1:

File: test/test_all.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+        assert release["checksum"] == "SHA-256:" + hashlib.sha256(release_archive_path.read_bytes()).hexdigest()
+
+
+def check_logs(configuration, golden_logs_parent_path, logs_subpath):
+    """Run tests to determine whether the engine's logs are as expected.
+
+    Keyword arguments:
+    configuration -- dictionary defining the libraries-repository-engine configuration
+    golden_logs_parent_path -- parent path for the golden master logs to compare the actual logs against
+    golden_logs_parent_path -- sub-path for both the actual and golden master logs
```
Comment (silvanocerza) [inline test/test_all.py:129]:
```suggestion
    logs_subpath -- sub-path for both the actual and golden master logs
```

**Review by silvanocerza — CHANGES_REQUESTED** [review]
Everything looks great, just a small comment to edit to reach approval. :+1:

**Review by silvanocerza — APPROVED** [review]

File: test/test_all.py (hunk 3 above)
Comment (per1234) [reply test/test_all.py]: Thanks Silvano!

**Pull request merged**